pkg/api/v1/types.go

```diff
@@ -2165,6 +2165,14 @@ type PodProxyOptions struct {
 	Path string `json:"path,omitempty"`
 }
 
+// ServiceProxyOptions is the query options to a Service's proxy call.
+type ServiceProxyOptions struct {
```

Comment:
@feihujiang I'm so sorry no one got back to you on this. I'd love it if you could combine, but considering how close 1.2 is, I think I'm willing to accept this even with separate types, if you don't have time for doing the combination.